src/Bicep.LangServer/Completions/SyntaxMatcher.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Bicep.Core.Syntax;
+
+namespace Bicep.LanguageServer.Completions
+{
+    public static class SyntaxMatcher
+    {
+        public static bool IsTailMatch<T1, T2, T3>(IList<SyntaxBase> nodes, Func<T1, T2, T3, bool> predicate) 
```

Comment:
I will add more overloads that support between 1 and 5 type parameters as the matching rules get converted over time...

Reply:
Can even look into code generation although it's probably not worth it for something this simple.